libvips/conversion/bandjoin.c

```diff
@@ -116,15 +116,27 @@ vips_bandjoin_buffer( VipsBandarySequence *seq,
 		q1 = q;
 		p1 = p[i];
 
-		for( x = 0; x < width; x++ ) {
-			for( z = 0; z < ips; z++ )
-				q1[z] = p1[z];
+		if( ips == 1 ) {
+			for( x = 0; x < width; x++ ) {
+				q1[0] = p1[0];
```

Comment:
You could swap this for `p1[x]` and get rid of the `p1 += 1` below, it should save a little more.